File: CS/WindowsFormsApplication1/MyDockManager/MyCustomDrawArgs.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking.Paint;
using DevExpress.XtraBars.Styles;
using DevExpress.Utils;
using DevExpress.XtraBars.Docking;

namespace WindowsFormsApplication1
{
    public delegate void CustomDrawEventHandler(object sender, MyCustomDrawArgs e);

    public class MyCustomDrawArgs
    {

        public MyCustomDrawArgs(object args, DockPanel panel)
        {
            _Args = args;
            _Panel = panel;
            Handled = false;
        }

        private bool _Handled;
        public bool Handled
        {
            get { return _Handled; }
            set { _Handled = value; }
        }

        private object _Args;
        public object Args
        {
            get { return _Args; }
            set { _Args = value; }
        }

        private DockPanel _Panel;
        public DockPanel Panel
        {
            get { return _Panel; }
            set { _Panel = value; }
        }
    }


}

File: VB/WindowsFormsApplication1/MyDockManager/MyCustomDrawArgs.vb
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.XtraBars.Docking

Namespace WindowsFormsApplication1

    Public Delegate Sub CustomDrawEventHandler(ByVal sender As Object, ByVal e As MyCustomDrawArgs)

    Public Class MyCustomDrawArgs

        Public Sub New(ByVal args As Object, ByVal panel As DockPanel)
            _Args = args
            _Panel = panel
            Handled = False
        End Sub

        Private _Handled As Boolean

        Public Property Handled As Boolean
            Get
                Return _Handled
            End Get

            Set(ByVal value As Boolean)
                _Handled = value
            End Set
        End Property

        Private _Args As Object

        Public Property Args As Object
            Get
                Return _Args
            End Get

            Set(ByVal value As Object)
                _Args = value
            End Set
        End Property

        Private _Panel As DockPanel

        Public Property Panel As DockPanel
            Get
                Return _Panel
            End Get

            Set(ByVal value As DockPanel)
                _Panel = value
            End Set
        End Property
    End Class
End Namespace
